src/main/java/de/jenvy/haitatsu/packet/codec/HaitatsuPacketDecoder.java

```diff
@@ -1,25 +1,25 @@
 package de.jenvy.haitatsu.packet.codec;
 
-import de.jenvy.haitatsu.api.packet.HaitatsuPacket;
 import de.jenvy.haitatsu.api.packet.HaitatsuPacketFactory;
 import de.jenvy.haitatsu.packet.AbstractHaitatsuByteBuf;
 import io.netty.buffer.ByteBuf;
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.handler.codec.ByteToMessageDecoder;
+import org.jetbrains.annotations.NotNull;
 
 import java.util.List;
 
 public class HaitatsuPacketDecoder extends ByteToMessageDecoder {
 
     private final HaitatsuPacketFactory packetFactory;
 
-    public HaitatsuPacketDecoder(HaitatsuPacketFactory packetFactory) {
+    public HaitatsuPacketDecoder(@NotNull HaitatsuPacketFactory packetFactory) {
         this.packetFactory = packetFactory;
     }
 
     @Override
-    protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) {
-        HaitatsuPacket packet = this.packetFactory.getPacket(in.readInt());
+    protected void decode(@NotNull ChannelHandlerContext ctx, @NotNull ByteBuf in, @NotNull List<Object> out) {
+        var packet = this.packetFactory.getPacket(in.readInt());
```

Comment:
We can use the var keyword to not repeat the type over and over again. Also, this keeps all variables names ontop each other.